### PR TITLE
Curve25519 fixes

### PR DESCRIPTION
**Pull request by bigbrett**
Fixes curve25519 implementation, changing it to use new [wolfCrypt DER API ](https://github.com/wolfSSL/wolfssl/pull/8129) for ser/de instead of raw keys

### CODE REVIEW

**Comment by bigbrett** [comment]
Just noticed wolfCrypt tests are failing - didn't try that locally. I should turn that on by default. I'll investigate

File: src/wh_crypto.c [hunk 1]
```diff
@@ -36,6 +36,7 @@
 #include "wolfssl/wolfcrypt/types.h"
 #include "wolfssl/wolfcrypt/error-crypt.h"
 #include "wolfssl/wolfcrypt/asn.h"
+#include "wolfssl/wolfcrypt/asn_public.h"
```
Comment (billphipps) [inline src/wh_crypto.c:39]: asn.h include asn_public.h.  Were you getting a compile warning/error without asn_public.h explicitly included?

File: src/wh_crypto.c [hunk 2]
```diff
@@ -207,49 +208,54 @@ int wh_Crypto_EccUpdatePrivateOnlyKeyDer(ecc_key* key, uint16_t pub_size,
...
+WOLFSSL_API int wc_Curve25519PublicKeyDecode(
+    const byte* input, word32* inOutIdx, curve25519_key* key, word32 inSz);
+#endif
+#ifdef HAVE_CURVE25519_KEY_EXPORT
+WOLFSSL_API int wc_Curve25519PrivateKeyToDer(
+    curve25519_key* key, byte* output, word32 inLen);
+WOLFSSL_API int wc_Curve25519PublicKeyToDer(
+    curve25519_key* key, byte* output, word32 inLen, int withAlg);
+#endif
+#endif /* HAVE_CURVE25519 */
```
Comment (billphipps) [inline src/wh_crypto.c:225]: Aren't these already declared in asn_public.h?

**Review by billphipps — CHANGES_REQUESTED** [review]
Thanks for cleaning this up! I had a couple of questions about asn.h.

**Review by billphipps — DISMISSED** [review]
LGTM!

File: src/wh_crypto.c (hunk 2 above)
Comment (bigbrett) [reply src/wh_crypto.c]: yep, this should have been removed. fixed

File: src/wh_crypto.c (hunk 1 above)
Comment (bigbrett) [reply src/wh_crypto.c]: I'm dumb, removed

**Review by billphipps — APPROVED** [review]
Excellent!

**Pull request merged**